pkg/flowkit/util/mixpanel.go

```diff
@@ -0,0 +1,169 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+const (
+	MIXPANEL_TRACK_URL   = "https://api.mixpanel.com/track"
+	MIXPANEL_QUERY_URL   = "https://mixpanel.com/api/2.0/engage?project_id=2154593"
+	MIXPANEL_PROFILE_URL = "https://api.mixpanel.com/engage#profile-set"
+)
+
+var MIXPANEL_PROJECT_TOKEN = ""
+var MIXPANEL_SERVICE_ACCOUNT_SECRET = ""
+
+type MixpanelClient struct {
+	token   string
+	baseUrl string
+}
+
+func TrackCommandUsage(command *cobra.Command) error {
+	mixpanelEvent := newEvent(command)
+	mixpanelEvent.setUpEvent(MIXPANEL_PROJECT_TOKEN, FLOW_CLI)
+	eventPayload, err := encodePayload(mixpanelEvent)
+	if err != nil {
+		return err
+	}
+	payload := strings.NewReader(eventPayload)
+	req, err := http.NewRequest("POST", MIXPANEL_TRACK_URL, payload)
+	if err != nil {
+		return err
+	}
+	req.Header.Add("Accept", "text/plain")
+	req.Header.Add("Content-Type", "application/json")
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+	_, err = ioutil.ReadAll(res.Body)
+
+	if err != nil {
+		return err
+	}
+	if res.StatusCode >= 400 {
+		return fmt.Errorf("invalid response status code %d for tracking command usage", res.StatusCode)
+	}
+
+	return nil
+}
+
+func encodePayload(obj interface{}) (string, error) {
+	b, err := json.Marshal(obj)
+	if err != nil {
+		return "", err
+	}
+	formattedString := "[" + string(b) + "]"
```

Comment:
This is a bit sketchy. It's better if you pass an array of interfaces to Marshal and it will produce the array as output. Do it like `[]any{ obj }`

Reply:
addressed